src/ApiManagement/ApiManagement.ServiceManagement/Commands/NewAzureApiManagementDiagnostic.cs

```diff
@@ -0,0 +1,99 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using System;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Properties;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Models;
+
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementDiagnostic", SupportsShouldProcess = true)]
+    [OutputType(typeof(PsApiManagementDiagnostic))]
+    public class NewAzureApiManagementDiagnostic : AzureApiManagementCmdletBase
+    {
+        [Parameter(
+            ValueFromPipeline = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@solankisamir for `New-*` cmdlets, it's OK to keep the `ValueFromPipelineByPropertyName` property on parameters other than `-ResourceId` (since there's no `-ResourceId` parameter for these cmdlets)